skaha/src/main/java/org/opencadc/skaha/session/PostAction.java

```diff
@@ -244,28 +261,127 @@ public void doAction() throws Exception {
         if (requestType.equals(REQUEST_TYPE_APP)) {
             if (appID == null) {
                 // create an app
-                
+
                 // gather job parameters
                 String image = syncInput.getParameter("image");
-                List<String> params = syncInput.getParameters("param");
-                
+
                 if (image == null) {
                     throw new IllegalArgumentException("Missing parameter 'image'");
                 }
-                
-                attachDesktopApp(image, params);
-                
+
+                attachDesktopApp(image);
+
             } else {
                 throw new UnsupportedOperationException("Cannot modify an existing app.");
             }
         }
     }
-    
+
+    void ensureUserBase() throws IOException, InterruptedException {
+        final Path homeDir = Paths.get(String.format("%s/%s", this.homedir, this.userID));
+
+        if (Files.notExists(homeDir)) {
+            allocateUser();
+        }
+    }
+
+    void allocateUser() throws IOException, InterruptedException {
```

Comment:
Can you use SessionAction execute() to do some of the work in this method?  It would be good to keep the System call pattern in one place.